core/cosmosdb/cache-invalidator/src/main/scala/org/apache/openwhisk/core/database/cosmosdb/cache/CacheInvalidatorConfig.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.openwhisk.core.database.cosmosdb.cache
+
+import java.net.URI
+
+import com.microsoft.azure.documentdb.changefeedprocessor.{DocumentCollectionInfo => JDocumentCollectionInfo}
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigUtil.joinPath
+import pureconfig.loadConfigOrThrow
+
+import scala.concurrent.duration.FiniteDuration
+
+case class DocumentCollectionInfo(connectionInfo: ConnectionInfo, collectionName: String) {
+
+  def asJava: JDocumentCollectionInfo = {
+    val info = new JDocumentCollectionInfo
+    info.setUri(new URI(connectionInfo.endpoint))
+    info.setDatabaseName(connectionInfo.db)
+    info.setCollectionName(collectionName)
+    info.setMasterKey(connectionInfo.key)
+    info
+  }
+}
+
+case class ConnectionInfo(endpoint: String, key: String, db: String)
+
+case class FeedConfig(hostname: String, leaseCollection: String)
+
+case class InvalidatorConfig(port: Int, feedPublishTimeout: FiniteDuration, clusterId: Option[String])
+
+case class CacheInvalidatorConfig(globalConfig: Config) {
```

Comment:
Can we use pureconfig here to be consistent with other OW components?

Reply:
Yes. Purpose here is to adapt the config for test runs in a simplified manner. Our current approach in common and other module involve direct loading of global config which makes it very hard to manipulate the config for different test scenarios.

In this service I am trying to follow an approach where config is also passed down the line and hence not rely on a global singleton